server/server.go

```diff
@@ -1711,15 +1710,20 @@ func (s *Server) watchServicePrimaryAddrLoop(serviceName string) {
 	defer s.serverLoopWg.Done()
 	ctx, cancel := context.WithCancel(s.serverLoopCtx)
 	defer cancel()
+
 	serviceKey := fmt.Sprintf("/ms/%d/%s/%s/%s", s.clusterID, serviceName, fmt.Sprintf("%05d", 0), "primary")
-	// TODO: need to refactor after we redefine the member
-	leader := &pdpb.Member{}
-	ok, _, err := etcdutil.GetProtoMsgWithModRev(s.client, serviceKey, leader)
+	log.Info("start to watch", zap.String("service-key", serviceKey))
+
+	primary := &tsopb.Participant{}
+	ok, _, err := etcdutil.GetProtoMsgWithModRev(s.client, serviceKey, primary)
 	if err != nil {
 		log.Error("get service primary addr failed", zap.String("service-key", serviceKey), zap.Error(err))
 	}
-	if err == nil && ok {
-		s.servicePrimaryMap.Store(serviceName, leader.GetName())
+	listenUrls := primary.GetListenUrls()
+	if ok && len(listenUrls) > 0 {
+		s.servicePrimaryMap.Store(serviceName, listenUrls[0])
```

Comment:
where it is ensured that use [0] as primary?

Reply:
When calling new Participant. For now, we only save one url. using array is for extensibility in future.